### PR TITLE
[SPARK-33861][SQL][FOLLOWUP] Simplify conditional in predicate should consider deterministic 

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This pr address https://github.com/apache/spark/pull/30865#pullrequestreview-562344089 to fix simplify conditional in predicate should consider deterministic.

### Why are the changes needed?

Fix bug.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicateSuite.scala [hunk 1]
```diff
@@ -158,6 +158,15 @@ class SimplifyConditionalsInPredicateSuite extends PlanTest {
     testProjection(originalCond, expectedExpr = originalCond)
   }
 
+  test("CASE WHEN non-deterministic-cond THEN false END") {
+    val originalCond =
+      CaseWhen(Seq((UnresolvedAttribute("i") > Rand(0), FalseLiteral)))
+    val expectedCond = And(UnresolvedAttribute("i") > Rand(0), FalseLiteral)
+    // nondeterministic expressions are only allowed in Project, Filter, Aggregate or Window,
+    testFilter(originalCond, expectedCond = FalseLiteral)
```
Comment (wangyum) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicateSuite.scala:166]: `expectedCond = FalseLiteral` cause by https://github.com/apache/spark/blob/26d8df300a1a57e220b1a0f9814795f68101f28b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala#L295-L298

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala [hunk 2]
```diff
@@ -64,12 +64,8 @@ object SimplifyConditionalsInPredicate extends Rule[LogicalPlan] {
       And(cond, trueValue)
     case CaseWhen(Seq((cond, trueValue)), Some(TrueLiteral)) =>
       Or(Not(cond), trueValue)
-    case CaseWhen(Seq((_, FalseLiteral)), Some(FalseLiteral) | None) =>
-      FalseLiteral
     case CaseWhen(Seq((cond, FalseLiteral)), Some(elseValue)) =>
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala:67]: This case has handled by https://github.com/apache/spark/blob/3ceffeb42a69365ec811cf09119f995eabe68954/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala#L62-L64

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala [hunk 3]
```diff
@@ -64,12 +64,8 @@ object SimplifyConditionalsInPredicate extends Rule[LogicalPlan] {
...
       And(Not(cond), elseValue)
-    case CaseWhen(Seq((cond, TrueLiteral)), Some(FalseLiteral) | None) =>
-      cond
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala:72]: This case has handled by https://github.com/apache/spark/blob/3ceffeb42a69365ec811cf09119f995eabe68954/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala#L62-L64

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala [hunk 4]
```diff
@@ -64,12 +64,8 @@ object SimplifyConditionalsInPredicate extends Rule[LogicalPlan] {
       And(cond, trueValue)
     case CaseWhen(Seq((cond, trueValue)), Some(TrueLiteral)) =>
       Or(Not(cond), trueValue)
-    case CaseWhen(Seq((_, FalseLiteral)), Some(FalseLiteral) | None) =>
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalsInPredicate.scala:67]: can we update the classdoc of this rule?

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]
LGTM thanks for handling this

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**